### PR TITLE
Feature/demo

### PR DESCRIPTION
**Pull request by MuskettaMan**
### Description

Added some demo assets into our world and resolved some small issues with shadows and PBR. PBR still looks broken on flat surfaces, which has to be resolved later.

### CODE REVIEW

File: engine/renderer/private/gbuffers.cpp [hunk 1]
```diff
@@ -15,10 +15,15 @@ GBuffers::GBuffers(const std::shared_ptr<GraphicsContext>& context, glm::uvec2 s
...
-    _shadowFormat = supportedDepthFormat.value();
+    if (supportedDepthFormat.has_value())
+    {
+        _depthFormat = supportedDepthFormat.value();
+        _shadowFormat = supportedDepthFormat.value();
+    }
+    else
+    {
+        assert(false && "No supported depth format!");
+    }
```
Comment (mmzala) [inline engine/renderer/private/gbuffers.cpp:26]: Why change this to an if statement when the assert exits the program anyway when there is no value available? Couldn't this just have been left as previously, like this:
`assert(supportedDepthFormat.has_value() && "No supported depth format!");`

File: engine/renderer/private/pipelines/skydome_pipeline.cpp [hunk 2]
```diff
@@ -54,7 +54,8 @@ void SkydomePipeline::RecordCommands(vk::CommandBuffer commandBuffer, uint32_t c
     colorAttachmentInfos[0].imageView = _context->Resources()->ImageResourceManager().Access(_hdrTarget)->views[0];
     colorAttachmentInfos[0].imageLayout = vk::ImageLayout::eAttachmentOptimalKHR;
     colorAttachmentInfos[0].storeOp = vk::AttachmentStoreOp::eStore;
-    colorAttachmentInfos[0].loadOp = vk::AttachmentLoadOp::eLoad;
+    colorAttachmentInfos[0].loadOp = vk::AttachmentLoadOp::eLoad; // eClear;
+    colorAttachmentInfos[0].clearValue.color = vk::ClearColorValue { .float32 = { { 0.0f, 0.0f, 0.0f, 0.0f } } };
```
Comment (mmzala) [inline engine/renderer/private/pipelines/skydome_pipeline.cpp:58]: Clear color is not needed

File: engine/source/pipelines/skydome_pipeline.cpp
Comment (mmzala) [inline engine/source/pipelines/skydome_pipeline.cpp:1]: File should be removed

File: engine/renderer/private/pipelines/geometry_pipeline.cpp [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #include "pipelines/geometry_pipeline.hpp"
 
+#include "../vulkan_helper.hpp"
```
Comment (mmzala) [inline engine/renderer/private/pipelines/geometry_pipeline.cpp:3]: Can be removed as it is already included at the bottom of includes

File: external/imgui_entt_entity_editor/imgui_entt_entity_editor.hpp [hunk 4]
```diff
@@ -1,9 +1,9 @@
 // for the license, see the end of the file
 #pragma once
 
+#include <functional>
```
Comment (mmzala) [inline external/imgui_entt_entity_editor/imgui_entt_entity_editor.hpp:4]: Was this an automatic change or an accident? The change is not really needed

**Review by mmzala — CHANGES_REQUESTED** [review]
Nice, left a few small comments. Also why is wren added to this PR?

File: engine/renderer/private/gbuffers.cpp (hunk 1 above)
Comment (MuskettaMan) [reply engine/renderer/private/gbuffers.cpp:26]: code linting fix 😭

File: external/imgui_entt_entity_editor/imgui_entt_entity_editor.hpp (hunk 4 above)
Comment (MuskettaMan) [reply external/imgui_entt_entity_editor/imgui_entt_entity_editor.hpp:4]: cause clang tidy sorts the includes

**Review by mmzala — APPROVED** [review]
Ah, so it was clang tidy that changed external files. I didn't think about that. LGTM now

**Pull request merged**